vfx/gl/texture.cc

```diff
@@ -0,0 +1,81 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "vfx/gl/texture.h"
+
+#include "ui/gl/gl_bindings.h"
+
+namespace vfx {
+
+Texture::Desc::Desc()
+  : internal_format(GL_RGBA),
+    format(GL_RGBA),
+    type(GL_UNSIGNED_BYTE) { 
+}
+
+Texture::Texture() : id_(0) {
+}
+
+Texture::Texture(const Desc& desc) : id_(0), size_(desc.size) {
+  glGenTextures(1, &id_);
+  glBindTexture(GL_TEXTURE_2D, id_);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  glTexImage2D(GL_TEXTURE_2D, 0, desc.internal_format, desc.size.width(),
+               desc.size.height(), 0, desc.format, desc.type, nullptr);
+}
+
+Texture::~Texture() {
+  DeleteTexture();
+}
+
+Texture Texture::CreateRGBA(gfx::Size size) {
+  Desc desc;
+  desc.size = std::move(size);
+  desc.internal_format = GL_RGBA;
+  desc.format = GL_RGBA;
+  desc.type = GL_UNSIGNED_BYTE;
+  return Texture(desc);
+}
+
+Texture Texture::CreateDepth(gfx::Size size) {
+  Desc desc;
+  desc.size = std::move(size);
+  desc.internal_format = GL_DEPTH_COMPONENT;
```

Comment:
GL_DEPTH_COMPONENT is [not core in GLES2](https://www.khronos.org/opengles/sdk/docs/man/), it requires the [OES_depth_texture](https://www.khronos.org/registry/gles/extensions/OES/OES_depth_texture.txt) extension. I don't know if you're guaranteed that by whatever GL library this code runs on top of, but you might want to check for the extension before you use this